### PR TITLE
Add basic support to monthly recurrence

### PR DESCRIPTION
**Pull request by bruteforcecat**
Implement basic monthly and days_of_month support.

It also support negative day_of_month as described in RFC. There are examples in the test cases.

This is a basic implementation without changing code structure and I am not 100% understanding the whole codebase. So please do let me know if there is any issue you found.

### CODE REVIEW

File: lib/cocktail/validation/day_of_month.ex [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  defp do_normalize_day_of_month(day_of_month, days_in_month) when day_of_month > days_in_month do
+    days_in_month
+  end
+
+  defp do_normalize_day_of_month(day_of_month, _days_in_month) when day_of_month > 0 do
+    day_of_month
+  end
+
+  defp do_normalize_day_of_month(day_of_month, days_in_month) when -day_of_month > days_in_month do
```
[the remark this answers is not in the document]
Comment (bruteforcecat) [reply lib/cocktail/validation/day_of_month.ex]: we should probably ignore this warning. Otherwise it will break credo & mix format, if a space is added

**Comment by doughsay** [comment]
Great stuff; thanks so much for your contribution!  The sourcelevel failure seems to be a false positive, so we can ignore that one, but to maintain our 100% test coverage it would be nice to figure out how to get that last missed line, or remove it if it's not needed.

running `mix coveralls.html` gives you html output of the coverage (because codecov.io seems a bit broken and isn't clearly showing what's missing).  Here's the missing line:

Screen Shot 2020-01-06 at 10 20 20 AM

Is it quick and easy to add a test to cover this case?

**Comment by bruteforcecat** [comment]
@doughsay Thanks for the kind words. I added test case :)

**Comment by doughsay** [comment]
superseded by #138

**Pull request closed without merging**